arch/x86/core/ia32/swap.S

```diff
@@ -303,6 +303,20 @@ CROHandlingDone:
 
 	movl    %eax, _kernel_offset_to_current(%edi)
 
+#if defined(CONFIG_THREAD_LOCAL_STORAGE) && defined(CONFIG_GDT_DYNAMIC)
+	pushl	%eax
+
+	/* Push EAX (thread pointer) as argument to z_x86_tls_update_gdt() */
+	pushl	%eax
+	call	z_x86_tls_update_gdt
```

Comment:
I'm wondering if we could create a generic C-level swap hook where we currently call `z_x86_swap_update_page_tables` and invoke from there

Reply:
having to get back to a C ABI here would be expensive when for most architectures the only operation required here is to set a register. If the x86-32 code is modified to just set the GS base address through the MSR, that would require only a couple of instructions here.

Reply:
as mentioned elsewhere we're keeping strict 32-bit CPU compatibility for the 32-bit x86 arch.